cli/flox/src/commands/environment.rs

```diff
@@ -1380,20 +1386,101 @@ impl Pull {
 
         let pointer_content =
             serde_json::to_string_pretty(&pointer).context("Could not serialize pointer")?;
-        let pointer_path = dot_flox_path.join(ENVIRONMENT_POINTER_FILENAME);
 
-        fs::create_dir_all(&dot_flox_path).context("Could not create .flox/ directory")?;
+        let dot_flox_parent = dot_flox_path
+            .parent()
+            .context("Could not get .flox/ parent")?;
+        fs::create_dir_all(dot_flox_parent).context("Could not create .flox/ parent directory")?;
+
+        let temp_dot_flox_dir = tempfile::TempDir::with_prefix_in(DOT_FLOX, dot_flox_parent)
+            .context("Could not create temporary directory for cloning environment")?;
```

Comment:
```suggestion
        // Pulls the environment into a temp directory which is later renamed to `.flox`.
        // We do this to avoid populating the floxmeta branch `owner/name.encode(path to .flox)`
        // in case building fails or the user aborts the fixup.
        // The branch will be adjusted when the environment is opened the next time
        // by the `ensure_branch` routine.
        let temp_dot_flox_dir = tempfile::TempDir::with_prefix_in(DOT_FLOX, dot_flox_parent)
            .context("Could not create temporary directory for cloning environment")?;
```

Please don't commit yet. 
Still thinking if there's an edge case where we can't depend on automatic adjustment